### ISSUE TITLE
[RFC] Ensure display name uniqueness across the entire platform

### ISSUE THREAD

**Issue by jsonnull** [jsonnull/aleamancer#9]
I see three major ways forward:

1. Allow duplicate display name text and use a tag to differentiate between duplicates in the client. Discord does this. Looks like Jason#1234.

2. Do not allow duplicate display name text and make display names first-come-first-serve. Using this strategy, we can still allow users to have a custom display name per game they are a member of. A complication with this strategy is that we would also have to consider allowing users to have a custom photo to accompany each username, which I do not think would scale well.

3. Just allow duplicate display names.

I'm looking forward to comments and further ideas.

**Comment by krisaoe** [comment]
For scalability, I think #1.
Works well on Discord, avoids potential issues arising from confusion etc.

**Comment by jsonnull** [comment]
I agree that will scale well. I also like that it allows duplicate display names without issue, meaning nobody doesn't get the name they want because it's held by another account.
